docs: Update Azure migration guide styling (#147)
infracost/docs: docs/guides/azure_devops_migration.md

## What's new?

> 🚀 The new [Infracost Azure DevOps Extension](https://marketplace.visualstudio.com/items?itemName=Infracost.infracost-tasks) provides a set of Azure Pipelines tasks offering a composable way of using Infracost in your pipeline. These JavaScript tasks simplify integrating Infracost into your Azure Pipelines. In addition, we've added Azure specific output formats, a cost summary table, and different behaviors, so you can control when comments are posted.

Reviewer: ```suggestion
🚀 The new [Infracost Azure DevOps Extension](https://marketplace.visualstudio.com/items?itemName=Infracost.infracost-tasks) provides a set of Azure Pipelines tasks offering a composable way of using Infracost in your pipeline. These JavaScript tasks simplify integrating Infracost into your Azure Pipeline. In addition, we've added Azure specific output formats, a cost summary table, and different behaviors, so you can control when comments are posted.
```
Product name: `Azure Pipelines` but a single entity is  `Pipeline`

Author: > This article is a detailed reference guide to Azure Pipelines YAML pipelines

It's the first line of their documentation, looks mouthful to me :) Do we want to capitalize the "Pipeline" then? It's not the name anymore.

Reviewer: yeah might be best just to lowercase it

### Composable tasks

The [Azure DevOps Extension](https://marketplace.visualstudio.com/items?itemName=Infracost.infracost-tasks) contains two main tasks:

> - InfracostSetup: installs the Infracost CLI in your Azure Pipelines.

Reviewer: ```suggestion
- InfracostSetup: install the Infracost CLI in your Azure Pipeline.
```

Yes the product might be called `Azure Pipelines` but you install a task into a single **pipeline**

3. Versioning for the CI integration: Each Infracost task has a version, `InfracostSetup@0` & `InfracostComment@0`, which also supports Semver. So you can use `@0` to get backward-compatible updates for the extension (e.g. bug fixes).

### CI-specific formats

The `infracost output` command now has two new format options: `azure-repos-comment` and `slack-message`. We already have formats for GitHub & GitLab which are used by our [GitHub Actions](https://github.com/infracost/actions) and [GitLab CI](https://gitlab.com/infracost/infracost-gitlab-ci/) integrations.

### Cost summary

As shown in the screenshot at the top of this page, comments now include a summary table showing the total cost diff for any projects that have changed.

### Comment behaviors

The InfracostComment task includes a `behavior` and a `targetType` attributes.

Behavior describes how and when comments should be posted; we support four options:

> - `update`: Create a single comment and update it on changes. This is the "quietest" option. For Azure DevOps Repos users, comments will simply be overwritten. GitHub users have additional UI that shows [what/when changed](https://docs.github.com/en/communities/moderating-comments-and-conversations/tracking-changes-in-a-comment) when the comment is updated. Pull request followers will only be notified on the comment create (not updates), and the comment will stay at the same location in the comment history.

Author: It seems like "the quietest option" is not true for Azure Repos. It sends an email notification with "base Build Service (infracost) has edited a comment" :)

- `delete-and-new`: Delete previous cost estimate comments and create a new one. Pull request followers will be notified on each comment.

> - `hide-and-new`: Minimize previous cost estimate comments and create a new one. Pull request followers will be notified on each comment. This behavior is available only for GitHub.

Reviewer: ```suggestion

```

Author: Do we want to remove it completely, even for GitHub users?

Reviewer: ah sorry I misunderstood - no let's keep it.

   - [Terragrunt](https://github.com/infracost/infracost-azure-devops/examples/terragrunt): a Terragrunt project
   - [Terraform Cloud/Enterprise](https://github.com/infracost/infracost-azure-devops/examples/terraform-cloud-enterprise): a Terraform project using Terraform Cloud/Enterprise
   - [Multi-project using config file](https://github.com/infracost/infracost-azure-devops/examples/multi-project/README.md#using-an-infracost-config-file): multiple Terraform projects using the Infracost [config file](https://www.infracost.io/docs/multi_project/config_file)

>    - [Multi-project using ajob matrix](https://github.com/infracost/infracost-azure-devops/examples/multi-project/README.md#using-azure-devops-pipeline-matrix-strategy): multiple Terraform projects using Azure Pipelines job matrix strategy

Reviewer: ```suggestion
   - [Multi-project using ajob matrix](https://github.com/infracost/infracost-azure-devops/examples/multi-project/README.md#using-azure-devops-pipeline-matrix-strategy): multiple Terraform projects using Azure Pipeline job matrix strategy
```
